pkg/providers/imagefamily/resolver.go

```diff
@@ -188,23 +194,38 @@ func prepareKubeletConfiguration(ctx context.Context, instanceType *cloudprovide
 	return kubeletConfig
 }
 
-func getSupportedImages(familyName *string) []DefaultImageOutput {
+func getSupportedImages(familyName *string, kubernetesVersion string) []DefaultImageOutput {
 	// TODO: Options aren't used within DefaultImages, so safe to be using nil here. Refactor so we don't actually need to pass in Options for getting DefaultImage.
-	imageFamily := getImageFamily(familyName, nil)
+	imageFamily := getImageFamily(familyName, kubernetesVersion, nil)
 	return imageFamily.DefaultImages()
 }
 
-func getImageFamily(familyName *string, parameters *template.StaticParameters) ImageFamily {
+func getImageFamily(familyName *string, kubernetesVersion string, parameters *template.StaticParameters) ImageFamily {
 	switch lo.FromPtr(familyName) {
 	case v1beta1.Ubuntu2204ImageFamily:
 		return &Ubuntu2204{Options: parameters}
 	case v1beta1.AzureLinuxImageFamily:
+		if UseAzureLinux3(kubernetesVersion) {
+			return &AzureLinux3{Options: parameters}
+		}
 		return &AzureLinux{Options: parameters}
 	default:
 		return &Ubuntu2204{Options: parameters}
 	}
 }
 
+// UseAzureLinux3 checks if the Kubernetes version is 1.32.0 or higher,
+// which is when Azure Linux 3 support starts
+func UseAzureLinux3(kubernetesVersion string) bool {
```

Comment:
nit: would prefer this in a `utils.go` file in the package, rather than adding it to the `resolver.go`